packages/react/polymorphic/src/polymorphic.test.tsx

```diff
@@ -40,7 +40,7 @@ export function ExtendedButtonUsingReactUtilsWithInternalInlineAs(
   props: React.ComponentProps<typeof Button>
 ) {
   /* Should not error with inline `as` component */
-  return <Button as={(props) => <button {...props} />} {...props} />;
+  return <Button as={(prop: any) => <button {...props} />} {...props} />;
```

Comment:
I spent a while trying to type these props for the consumer but it wasn't as easy as it sounds so parked it. Before they were typed as `any` but now they consumer needs explicitly type them.

Inline `as` prop functions are pretty rare so I'm not too worried about this. Hoping we can wait for feedback from consumers before we invest more time on it.

Reply:
I've just noticed I've done `prop` instead of `props` 🙈 I'll fix that later